controllers/secret_controller.go

```diff
@@ -220,25 +251,44 @@ func (r *SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-// createPagerdutyRoute creates an AlertManager Route for PagerDuty in memory.
-func createPagerdutyRoute(namespaceList []string) *alertmanager.Route {
+func createSubroutes(reqLogger logr.Logger, namespaceList []string, receiverType string) *alertmanager.Route{
+
+	var receiverCommon, receiverCritical, receiverError, receiverWarning, receiverDefault string
+
+	switch receiverType {
+	case "goalert":
```

Comment:
Oh one more nit.. can we make "goalert" and "pagerduty" constants, and then replace these strings with that constant wherever they appear in the file?

Reply:
Fixed by using enum